crypto/signer.go

```diff
@@ -108,13 +108,15 @@ type PublicKey interface {
 // serialized signature and doesn't need a private key. Use it when you already
 // have an appropriate signature (calculated elsewhere, not by SDK code), but
 // want to attach it to some structure/request.
+// Deprecated: construct [Signature] instead.
```

Comment:
@smallhive, I know you're using `StaticSigner` in some code, can you evaluate the change here?

Reply:
REST gateway is using it too, btw.

Reply:
REST worked without StatisSigner [before](https://github.com/nspcc-dev/neofs-rest-gw/commit/99a1e48b1d11af79d07f382692d72843c90d3228#diff-75efc19d08faf314e0ced46e860632366bbbc53735c01feae0397e40624b1480R528). I consider we may just revert some changes.

About another place, it works in the same way, only for the session token signing process. We already have required instruments to avoid `StaticSigner` using as well